pkgs/top-level/java-packages.nix

```diff
@@ -46,21 +65,25 @@ with pkgs;
       openjdk17-bootstrap = temurin-bin.jdk-17;
 
       temurin-bin = recurseIntoAttrs (
-        callPackage (
-          if stdenv.hostPlatform.isLinux then
-            ../development/compilers/temurin-bin/jdk-linux.nix
-          else
-            ../development/compilers/temurin-bin/jdk-darwin.nix
-        ) { }
+        let
+          temurinLinux = callPackage ../development/compilers/temurin-bin/jdk-linux.nix { };
+          temurinDarwin = callPackage ../development/compilers/temurin-bin/jdk-darwin.nix { };
+        in
+        if stdenv.hostPlatform.isLinux then
+          (mergeMetaPlatforms temurinLinux temurinDarwin)
+        else
+          (mergeMetaPlatforms temurinDarwin temurinLinux)
```

Comment:
This doesn't work as intended, because of `recurseIntoAttrs`; below also

```
nix-repl> legacyPackages.x86_64-linux.semeru-bin.meta.platforms                                                                 
[
  "aarch64-linux"
  "x86_64-linux"
]

nix-repl> legacyPackages.x86_64-darwin.semeru-bin.meta.platforms
[
  "aarch64-darwin"
  "x86_64-darwin"
]
```

Reply:
fixed it by mapping through the attrs members

```
nix-repl> temurin-bin.meta.platforms
[
  "aarch64-linux"
  "powerpc64le-linux"
  "riscv64-linux"
  "x86_64-linux"
  "aarch64-darwin"
  "x86_64-darwin"
]

nix-repl> temurin-bin-23.meta.platforms
[
  "aarch64-linux"
  "powerpc64le-linux"
  "riscv64-linux"
  "x86_64-linux"
  "aarch64-darwin"
  "x86_64-darwin"
]
```